### PR TITLE
Improve capzcrs CR merging

### PR DESCRIPTION
**Pull request by tuommaki**
Generic implementation doesn't preserve original CR spec fields. This is needed
to maintain dynamically managed subnets in AzureCluster.

### CODE REVIEW

**Comment by tuommaki** [comment]
When testing on `ghost` the subnets are now preserved.

**Review by fiunchinho — APPROVED** [review]
LGTM

File: service/controller/resource/capzcrs/create_test.go [hunk 1]
```diff
@@ -114,6 +125,76 @@ func Test_AzureConfigCRMapping(t *testing.T) {
...
+						Subnets: []*capzv1alpha3.SubnetSpec{
+							&capzv1alpha3.SubnetSpec{
+								ID:   "nodepool-np201",
+								Role: capzv1alpha3.SubnetNode,
+							},
+						},
+					},
+				},
+			},
+			desired: &capzv1alpha3.AzureCluster{},
```
Comment (nprokopic) [inline service/controller/resource/capzcrs/create_test.go:157]: Would it be useful to have a test case where there are, let's say, 2 desired and 3 original? An example where there is a diff, but both are non-null?

File: service/controller/resource/capzcrs/create.go [hunk 2]
```diff
@@ -290,6 +294,109 @@ func (r *Resource) updateCRs(ctx context.Context, desiredCRs []runtime.Object) e
...
+	}
+
+	if !cmp.Equal(d, o, cmpopts.IgnoreTypes(&metav1.ObjectMeta{}), cmpopts.IgnoreTypes(&metav1.TypeMeta{}, cmpopts.IgnoreTypes(&capzv1alpha3.Subnets{}))) {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func mergeAzureCluster(orig, desired runtime.Object) (runtime.Object, error) {
```
Comment (nprokopic) [inline service/controller/resource/capzcrs/create.go:316]: The func will always take subnets from original and set them to desired, right? There isn't an use case where subnets from both original and desired will be taken?

File: service/controller/resource/capzcrs/create_test.go (hunk 1 above)
Comment (tuommaki) [reply service/controller/resource/capzcrs/create_test.go:157]: Mapping from `AzureConfig` -> `AzureCluster` doesn't recognize subnets at all. Therefore desired would always have nil for subnets. So: No, it wouldn't be useful :slightly_smiling_face:

File: service/controller/resource/capzcrs/create.go (hunk 2 above)
Comment (tuommaki) [reply service/controller/resource/capzcrs/create.go:316]: Nope. More details above :slightly_smiling_face:

File: service/controller/resource/capzcrs/create_test.go (hunk 1 above)
Comment (nprokopic) [reply service/controller/resource/capzcrs/create_test.go:157]: Thanks!

**Review by nprokopic — APPROVED** [review]

**Pull request merged**